### PR TITLE
Issue #7638: Update doc for EqualsHashCode

### PR DESCRIPTION
**Pull request by DXTkastb**
**Issue #7638**
EqualsHashCode Check Doc Updation.
up

**Output of Example:**

```
kaustubh@dxtk:~$ cat Test.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="EqualsHashCode"/>
    </module>
</module>


kaustubh@dxtk:~$ cat Test.java
public static class Example1 {
    public int hashCode() {
        // code
    }
    public boolean equals(String o) { // violation, overloaded implementation of 'equals'
        // code
    }
}
public static class Example2 {
    public boolean equals(Object o) { // violation, no 'hashCode'
        // code
    }
    public boolean equals(String o) {
        // code
    }
}
public static class Example3 {
    public int hashCode() {
        // code
    }
    public boolean equals(Object o) { // OK
        // code
    }
    public boolean equals(String o) {
        // code
    }
}
public static class Example4 {
    public int hashCode() {
        // code
    }
    public boolean equals(java.lang.Object o) { // OK
        // code
   }
}
public static class Example5 {
    public static int hashCode(int i) {
        // code
    }
    public boolean equals(Object o) { // violation, overloaded implementation of 'hashCode'
        // code
    }
}
public static class Example6 {
    public int hashCode() { // violation, overloaded implementation of 'equals'
        // code
    }
    public static boolean equals(Object o, Object o2) {
        // code
    }
}


kaustubh@dxtk:~$ java -jar checkstyle-8.30-all.jar -c Test.xml Test.java
Starting audit...
[ERROR] /home/kaustubh/Test.java:2:5: Definition of 'hashCode()' without corresponding definition of 'equals()'. [EqualsHashCode]
[ERROR] /home/kaustubh/Test.java:11:5: Definition of 'equals()' without corresponding definition of 'hashCode()'. [EqualsHashCode]
[ERROR] /home/kaustubh/Test.java:44:5: Definition of 'equals()' without corresponding definition of 'hashCode()'. [EqualsHashCode]
[ERROR] /home/kaustubh/Test.java:50:5: Definition of 'hashCode()' without corresponding definition of 'equals()'. [EqualsHashCode]
Audit done.
Checkstyle ends with 4 errors.
```
@strkkk Please review my pull request and suggest necessary changes. I have taken these examples from [https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/equalshashcode/InputEqualsHashCodeEqualsParameter.java](url)

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java [hunk 1]
```diff
@@ -50,6 +50,62 @@
  * &lt;module name=&quot;EqualsHashCode&quot;/&gt;
  * </pre>
  *
+ * <p>Example:</p>
+ *
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java]: no need for empty line

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java [hunk 2]
```diff
@@ -50,6 +50,62 @@
...
+ * <pre>
+ * public static class Example1 { // violation, no correct `equals`
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java]: violation comment should be placed on line where violation occurs

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java [hunk 3]
```diff
@@ -50,6 +50,62 @@
...
+ *     public int hashCode() {
+ *         //code
+ *     }
+ *     public boolean equals(String o) {
+ *         //code
+ *     }
+ * }
+ * public static class Example2 { // violation, no `hashCode`
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java]: here too

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java [hunk 4]
```diff
@@ -50,6 +50,62 @@
...
+ *     public boolean equals(Object o) {
+ *         //code
+ *     }
+ *     public boolean equals(String o) {
+ *         //code
+ *     }
+ * }
+ * public static class Example3 { // no violation
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java]: should be `// OK` and on correct line

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java [hunk 5]
```diff
@@ -50,6 +50,62 @@
...
+ *         //code
+ *     }
+ *     public boolean equals(Object o) {
+ *         //code
+ *     }
+ *     public boolean equals(String o) {
+ *         //code
+ *     }
+ * }
+ * public static class Example4 { // no violation
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java]: same here

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java [hunk 6]
```diff
@@ -50,6 +50,62 @@
...
+ *     public int hashCode() {
+ *         //code
+ *     }
+ *     public boolean equals(java.lang.Object o) {
+ *         //code
+ *    }
+ * }
+ * public static class Example5 { // violation, no `hashCode` implementation
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java]: here too

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java [hunk 7]
```diff
@@ -50,6 +50,62 @@
...
+ *     public static int hashCode(int i) {
+ *         //code
+ *     }
+ *     public boolean equals(Object o) {
+ *         //code
+ *     }
+ * }
+ * public static class Example6 { // violation, no `equals` implementation
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java]: here too

**Comment by DXTkastb** [comment]
@strkkk can you please help me out, my pr fails one of the checks: wercker/build. It shows 401 when I click for details. Is it due to some connection issue? Or should I make changes to the PR to resolve this problem?

**Comment by strkkk** [comment]
@DXTkastb wercker failure is not related with your code, it can be ignored for now.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java [hunk 8]
```diff
@@ -49,7 +49,60 @@
  * <pre>
  * &lt;module name=&quot;EqualsHashCode&quot;/&gt;
  * </pre>
- *
+ * <p>Example:</p>
+ * <pre>
+ * public static class Example1 {
+ *     public int hashCode() {
+ *         //code
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java]: `//code` please add space -> `// code`
In lines below as well

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java [hunk 9]
```diff
@@ -49,7 +49,60 @@
...
+ *     }
+ *     public boolean equals(String o) { // violation, no correct `equals`
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java]: `equals` please use single quotes here and below

Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java]: `no correct` -> overloaded implementation of equals

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java [hunk 10]
```diff
@@ -49,7 +49,60 @@
...
+ * }
+ * public static class Example2 {
+ *     public boolean equals(Object o) { // violation, no `hashCode`
+ *         //code
+ *     }
+ *     public boolean equals(String o) {
+ *         //code
+ *     }
+ * }
+ * public static class Example3 { // ok
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java]: `// ok` -> // OK

in other cases below as well

Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java]: Please move `// ok` comments to methods, to make it similar to violation comments

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java [hunk 11]
```diff
@@ -49,7 +49,60 @@
...
+ *         //code
+ *     }
+ *     public boolean equals(Object o) {
+ *         //code
+ *     }
+ *     public boolean equals(String o) {
+ *         //code
+ *     }
+ * }
+ * public static class Example4 { // ok
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java]: here too, move to methods

**Comment by DXTkastb** [comment]
@strkkk I have made the changes suggested by you. 
If anything else needs to be changed please let me know.

File: src/xdocs/config_coding.xml [hunk 12]
```diff
@@ -1015,6 +1015,60 @@ String nullString = null;
         <source>
 &lt;module name=&quot;EqualsHashCode&quot;/&gt;
         </source>
+ <p>Example:</p>
```
Comment (strkkk) [inline src/xdocs/config_coding.xml]: please fix indentation, it should be on same level as previous tag

File: src/xdocs/config_coding.xml [hunk 13]
```diff
@@ -1015,6 +1015,60 @@ String nullString = null;
...
+<source>
```
Comment (strkkk) [inline src/xdocs/config_coding.xml]: here too

File: src/xdocs/config_coding.xml [hunk 14]
```diff
@@ -1015,6 +1015,60 @@ String nullString = null;
...
+}
+public static class Example6 {
+    public int hashCode() { // violation, overloaded implementation of 'equals'
+        // code
+    }
+    public static boolean equals(Object o, Object o2) {
+        // code
+    }
+}
+</source>
```
Comment (strkkk) [inline src/xdocs/config_coding.xml]: here too

File: src/xdocs/config_coding.xml [hunk 15]
```diff
@@ -1015,6 +1015,60 @@ String nullString = null;
...
+}
+public static class Example6 {
+    public int hashCode() { // violation, overloaded implementation of 'equals'
+        // code
+    }
+    public static boolean equals(Object o, Object o2) {
+        // code
+    }
+}
+       </source>
```
Comment (strkkk) [inline src/xdocs/config_coding.xml]: one more space is needed

**Comment by DXTkastb** [comment]
@strkkk I have updated the first post and squashed the commits.

**Comment by strkkk** [comment]
Ok to merge after CI pass

**Pull request merged**